geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-parquet/src/main/scala/org/locationtech/geomesa/parquet/io/SimpleFeatureReadSupport.scala

```diff
@@ -381,7 +381,7 @@ object SimpleFeatureReadSupport {
       val polys = Array.tabulate(x.polys.length) { i =>
         val shell = gf.createLinearRing(zip(x.polys(i).head, y.polys(i).head))
         val holes = if (x.polys(i).lengthCompare(1) == 0) { null } else {
-          Array.tabulate(x.polys(i).length - 1)(i => gf.createLinearRing(zip(x.polys(i)(i + 1), y.polys(i)(i + 1))))
+          Array.tabulate(x.polys(i).length - 1)(j => gf.createLinearRing(zip(x.polys(i)(j + 1), y.polys(i)(j + 1))))
```

Comment:
Was this a bug?

Reply:
Yeah, I opened 2690 for it, but it wasn't actually released anywhere yet, just in 2.4-SNAPSHOT

Reply:
I added some better unit tests that captured the fix, too.